producer/writer/consumer_service_writer.go

```diff
@@ -217,17 +217,22 @@ func (w *consumerServiceWriterImpl) Init(t initType) error {
 	}
 	if t == allowInitValueError {
 		if _, ok := err.(watch.InitValueError); ok {
+			w.logger.Warnf("invalid placement update: %v, continue to watch for placement updates", err)
 			return nil
 		}
 	}
-	return fmt.Errorf("m3msg consumer service writer init error: %v", err)
+	return fmt.Errorf("consumer service writer init error: %v", err)
 }
 
 func (w *consumerServiceWriterImpl) process(update interface{}) error {
 	var (
 		p         = update.(placement.Placement)
 		isSharded = p.IsSharded()
 	)
+	// Non sharded placement is only allowed for Shared consumption type.
```

Comment:
Should we make this an option? If someone (not us) wants to configure it this way, does m3msg want to support it?

Reply:
Hm I think it might still be useful in some cases, e.g., perhaps a user wants to have a RF=2 for backup purposes. Might be better to make this optional, which doesn't look like it's too difficult?